nexus_system_orchestrator/src/context.hpp

```diff
@@ -29,47 +29,267 @@
 #include <rclcpp_action/rclcpp_action.hpp>
 #include <rclcpp_lifecycle/lifecycle_node.hpp>
 
+#include <mutex>
+#include <optional>
+
 namespace nexus::system_orchestrator {
 
 class Context
 {
-public: using WorkOrder = common::WorkOrder;
-public: using WorkOrderGoalHandle =
+public:
+  using WorkOrder = common::WorkOrder;
+  using WorkOrderGoalHandle =
     rclcpp_action::ServerGoalHandle<nexus::endpoints::WorkOrderAction::ActionType>;
-public: using WorkcellTask = nexus_orchestrator_msgs::msg::WorkcellTask;
-public: using TaskState = nexus_orchestrator_msgs::msg::TaskState;
+  using WorkcellTask = nexus_orchestrator_msgs::msg::WorkcellTask;
+  using TaskState = nexus_orchestrator_msgs::msg::TaskState;
+
+  Context(rclcpp_lifecycle::LifecycleNode& node) : _node(node) {}
```

Comment:
Can we move the implementations of the constructor and member functions to a `context.cpp` translation unit?

Reply:
Done, https://github.com/osrf/nexus/pull/75/commits/1a073a5645b7fd2b0298ec130eb3845ed805dece